worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java

```diff
@@ -736,15 +738,15 @@ public Void call() throws Exception {
 
             int numFiles = -1;
             if (checkFilesize) {
-                File parentDir = new File(file.getParent());
-
-                for (File child : getFiles(rootDir, null, null)) {
-                    if (child.getName().endsWith(".schem") || child.getName().endsWith(".schematic")) {
-                        directorysizeKb += Files.size(Paths.get(child.getAbsolutePath())) / 1000.0;
-                        numFiles++;
+                List<File> toAddUp = getFiles(rootDir, null, null);
+                if(toAddUp != null &&  toAddUp.size() != 0) {
+                    for (File child : toAddUp) {
+                        if (child.getName().endsWith(".schem") || child.getName().endsWith(".schematic")) {
+                            directorysizeKb += Files.size(Paths.get(child.getAbsolutePath())) / 1000.0;
```

Comment:
1000 should be sufficient.

Reply:
it needs to be 1000.0 otherwise the result will remain an integer - even though it's being +='ed to a double - which I think is just silly but that's how it is :(